main.py

```
@@ -1,15 +1,73 @@
+            application = create_bot_app(settings)
+            logging.info(f"Computed WEBHOOK_URL={WEBHOOK_URL!r}")
+            await application.initialize()
+            await application.start()
+            
+            # Set webhook if public URL is provided; otherwise, ensure webhook is deleted
+            if WEBHOOK_URL:
+                try:
+                    await application.bot.set_webhook(url=WEBHOOK_URL, drop_pending_updates=True)
+                    logging.info("Webhook set successfully")
+                except Exception as e:
+                    logging.exception(f"Failed to set webhook: {e}")
```

Ruff (0.13.1)

Redundant exception object included in `logging.exception` call

(TRY401)